unit-tests/live/d400/test-depth-ae-toggle.py

```diff
@@ -0,0 +1,277 @@
+                
+                # Wait for AE to settle down and frames to stabilize
+                time.sleep(2.0)
+            
+            # Now wait for frames in current state
+            frames_before = frame_count[0]
+            timeout = 10.0  # Allow plenty of time for AE to settle and frames to arrive
+            start_time = time.time()
+            
+            log.d(f"Toggle {toggle_idx}: waiting for {frames_per_state} frames with AE={current_ae_state}, currently have {frame_count[0]} total")
+            
+            while (frame_count[0] - frames_before) < frames_per_state:
+                elapsed = time.time() - start_time
+                if elapsed > timeout:
+                    frames_received_this_iteration = frame_count[0] - frames_before
+                    log.e(f"Timeout waiting for frames in AE state {current_ae_state} (toggle {toggle_idx})")
+                    log.e(f"Waited {elapsed:.1f}s but only got {frames_received_this_iteration}/{frames_per_state} frames")
+                    log.e(f"Total frames collected: {frame_count[0]} (expected {num_toggles * frames_per_state})")
+                    try:
+                        depth_sensor.stop()
+                        depth_sensor.close()
+                    except RuntimeError:
+                        pass  # Sensor may have already stopped
+                    return False, frame_count[0], metadata_errors
+                time.sleep(0.01)
+            
+            log.d(f"Toggle {toggle_idx}: collected {frame_count[0] - frames_before} frames, total now {frame_count[0]}")
+            
+            # Check metadata for frames received in this state
+            for i in range(frames_before, frame_count[0]):
+                if i < len(frame_ae_metadata) and frame_ae_metadata[i] is not None:
+                    ae_metadata = frame_ae_metadata[i]
+                    ae_enabled_in_metadata = (ae_metadata != 0)
+                    if ae_enabled_in_metadata != current_ae_state:
+                        metadata_errors += 1
+                        log.d(f"Frame {i}: AE metadata mismatch - expected {current_ae_state}, metadata shows {ae_enabled_in_metadata}")
+    
+    finally:
+        try:
+            depth_sensor.stop()
+            depth_sensor.close()
+        except RuntimeError:
+            pass  # Sensor may have already stopped
+    
+    # Check for frame stalls by looking at timestamp gaps
+    if len(frame_timestamps) > 1:
+        for i in range(1, len(frame_timestamps)):
+            gap = frame_timestamps[i] - frame_timestamps[i-1]
+            if gap > max_frame_gap:
+                max_frame_gap = gap
+    
+    log.d(f"Received {frame_count[0]} total frames, max gap: {max_frame_gap:.3f}s, metadata errors: {metadata_errors}")
+    
+    # Success if we got enough frames and max gap is reasonable (< 1 second indicates no stall)
+    success = frame_count[0] >= (num_toggles * frames_per_state) and max_frame_gap < 1.0
```

Comment:
1 second max gap seems very long. Select profile FPS and verify not larger than expected + 2 ms delta.

Reply:
The gap does not measure the metadata change but frame drops. Frames should not drop while changing AE mode.